### ISSUE TITLE
populate glossary

### ISSUE THREAD

**Issue by tkphd** [hpc-carpentry/hpc-intro#37]
The glossary ([`reference.md`](../blob/gh-pages/reference.md)) needs content. For a start:

- [ ] [Accelerator](https://en.wikipedia.org/wiki/Hardware_acceleration)
- [ ] ~[Beowulf cluster](https://en.wikipedia.org/wiki/Beowulf_cluster)~
- [ ] [Central processing unit](https://en.wikipedia.org/wiki/Central_processing_unit)
- [ ] [Cloud computing](https://en.wikipedia.org/wiki/Cloud_computing)
- [ ] [Cluster](https://en.wikipedia.org/wiki/Computer_cluster)
- [ ] [Coupling, loose *vs.* tight](https://techdifferences.com/difference-between-loosely-coupled-and-tightly-coupled-multiprocessor-system.html)
- [ ] [CUDA](https://en.wikipedia.org/wiki/CUDA)
- [ ] [Distributed memory](https://en.wikipedia.org/wiki/Distributed_memory)
- [ ] [Flynn's taxonomy](https://en.wikipedia.org/wiki/Flynn%27s_taxonomy) ([SIMD](https://en.wikipedia.org/wiki/SIMD), [SIMT](https://en.wikipedia.org/wiki/Single_instruction,_multiple_threads))
- [ ] ~[General-purpose computing on graphics processing units](https://en.wikipedia.org/wiki/General-purpose_computing_on_graphics_processing_units)~
- [ ] [Graphics processing unit](https://en.wikipedia.org/wiki/Graphics_processing_unit)
- [ ] [Grid computing](https://en.wikipedia.org/wiki/Grid_computing)
- [ ] [Grid Engine](https://en.wikipedia.org/wiki/Oracle_Grid_Engine)
- [ ] ~[High availability computing](https://en.wikipedia.org/wiki/High_availability)~
- [ ] [High performance computing](https://en.wikipedia.org/w/index.php?title=High-performance_computing&redirect=no)
- [ ] [Hyper-Threading](https://en.wikipedia.org/wiki/Hyper-threading) ([SMT](https://en.wikipedia.org/wiki/Simultaneous_multithreading))
- [ ] [InfiniBand](https://en.wikipedia.org/wiki/InfiniBand)
- [ ] [Interconnect](https://en.wikipedia.org/wiki/Supercomputer_architecture)
- [ ] [Massively parallel](https://en.wikipedia.org/wiki/Massively_parallel)
- [ ] [Message passing interface](https://en.wikipedia.org/wiki/Message_Passing_Interface)
- [ ] [Node](https://en.wikipedia.org/wiki/Node_(computer_science))
- [ ] [OpenMP](https://en.wikipedia.org/wiki/OpenMP)
- [ ] [Parallel](https://en.wikipedia.org/wiki/Parallel_computing)
- [ ] [Serial](https://en.wikipedia.org/wiki/Serial_computer)
- [ ] [Server](https://en.wikipedia.org/wiki/Server_(computing))
- [ ] [Shared memory](https://en.wikipedia.org/wiki/Shared_memory)
- [ ] [Single instruction multiple data](https://en.wikipedia.org/wiki/SIMD)
- [ ] [Slurm](https://en.wikipedia.org/wiki/Slurm_Workload_Manager)
- [ ] [Symmetric multiprocessing](https://en.wikipedia.org/wiki/Symmetric_multiprocessing)
- [ ] [Simultaneous multithreading](https://en.wikipedia.org/wiki/Simultaneous_multithreading)
- [ ] [Supercomputer](https://en.wikipedia.org/wiki/Supercomputer) -- "[a major scientific instrument](https://www.hpcnotes.com/2015/10/essential-analogies-for-hpc-advocate.html)"
- [ ] [Workstation](https://en.wikipedia.org/wiki/Workstation)

**Comment by psteinb** [comment]
good idea and thanks for putting that into an issues. But always remember: Top priority for good teaching is *reduction*. There are many things in that list which I'd drop. I don't feel too polite to edit your issues directly. How about #38 ?

**Comment by ChristinaLK** [comment]
@tkphd I would say that grid and cloud computing are *not* out of scope -- I'd like to see this lessons adapted to use on a system like the Open Science Grid, which is not traditional HPC.  

I have an alternative proposal for forming the glossary.  How about going through the lessons and pulling out the terms that could use a definition?

**Comment by tkphd** [comment]
Good suggestion, @ChristinaLK.
Perhaps a hybrid approach: go through the existing lessons, tagging or linking instances of jargon to the glossary, then reviewing what's undefined and what's never referenced? Not a trivial task...

**Comment by ChristinaLK** [comment]
Yep. Reviewing the lessons seems like a manageable task and could be done piecemeal.

**Comment by psteinb** [comment]
It's all fine with me. Go ahead. I just fear that this repo will be 
restructured soon, so I suggest to hold your fingers for now.

**Comment by ChristinaLK** [comment]
Good point.  I'll focus on #40 and then come back to this.

**Comment by tkphd** [comment]
Reviewing this list, I'm honestly not sure the glossary is useful. When terms relevant to, and within scope for, the lessons come up, we define them. Anything that doesn't come up, is out-of-scope for us to define; a Web search is better suited than a glossary. Anything that does come up, but isn't adequately explained, we need to address within the lesson.

**Comment by mikerenfro** [comment]
Never noticed a glossary in the [shell-novice reference page](http://swcarpentry.github.io/shell-novice/reference/), but there it is. As a result, I don't think I'd abandon the glossary entirely, but it definitely shouldn't include terms we never use in the lessons (looks like the shell-novice glossary is keeping with that rule except maybe for "REPL"). Being able to point people at a single reference page including a review of terms seems useful, without them having to scroll through the entire lesson or hope for the best on a search. Not sure what else might go on a more complete reference page yet.

**Comment by psteinb** [comment]
some learners have asked for a glossary here and there during my HPC intro courses.

some automation would be helpful here I believe, so that if we define a term during a lesson, it is automagically appended to the glossary.

**Comment by annajiat** [comment]
I would like to vote to keep glossaries.
We can choose to link to the glossaries page from individual lesson materials.

**Comment by bkmgit** [comment]
May be helpful to have this, but could be in extra materials as done for [shell-lesson](http://swcarpentry.github.io/shell-novice/reference.html)

**Comment by tkphd** [comment]
It already is: Extras -> Reference.

**Comment by reid-a** [comment]
The glossary is no longer in reference.md, it's now in _extras/guide.md. Edited the leading comment on this issue to reflect the new location.
